integration/query_projection_compat_test.go

```diff
@@ -141,33 +145,55 @@ func TestQueryProjectionCompat(t *testing.T) {
 		"DotNotationInclude": {
 			filter:     bson.D{},
 			projection: bson.D{{"v.foo", true}},
-			skip:       "https://github.com/FerretDB/FerretDB/issues/2430",
 		},
 		"DotNotationIncludeTwo": {
 			filter:     bson.D{},
 			projection: bson.D{{"v.foo", true}, {"v.array", true}},
-			skip:       "https://github.com/FerretDB/FerretDB/issues/2430",
+		},
+		"DotNotationIncludeTwoReverse": {
+			filter:     bson.D{},
+			projection: bson.D{{"v.array", true}, {"v.foo", true}},
 		},
 		"DotNotationExclude": {
 			filter:     bson.D{},
 			projection: bson.D{{"v.foo", false}},
-			skip:       "https://github.com/FerretDB/FerretDB/issues/2430",
 		},
 		"DotNotationExcludeTwo": {
 			filter:     bson.D{},
 			projection: bson.D{{"v.foo", false}, {"v.array", false}},
-			skip:       "https://github.com/FerretDB/FerretDB/issues/2430",
 		},
 		"DotNotationExcludeSecondLevel": {
 			filter:     bson.D{},
 			projection: bson.D{{"v.array.42", false}},
-			skip:       "https://github.com/FerretDB/FerretDB/issues/2430",
 		},
 		"DotNotationIncludeExclude": {
 			filter:     bson.D{},
 			projection: bson.D{{"v.foo", true}, {"v.array", false}},
 			resultType: emptyResult,
-			skip:       "https://github.com/FerretDB/FerretDB/issues/2430",
+		},
+		"DotNotation5LevelInclude": {
+			filter:     bson.D{},
+			projection: bson.D{{"v.a.b.c.d", true}},
+		},
+		"DotNotation5LevelExclude": {
+			filter:     bson.D{},
+			projection: bson.D{{"v.a.b.c.d", false}},
+		},
+		"DotNotation4LevelInclude": {
+			filter:     bson.D{},
+			projection: bson.D{{"v.a.b.c", true}},
+		},
+		"DotNotation4LevelExclude": {
+			filter:     bson.D{},
+			projection: bson.D{{"v.a.b.c", false}},
+		},
+		"DotNotationArray": {
+			filter:     bson.D{},
+			projection: bson.D{{"v.array.0", true}},
+		},
+		"DotNotationArrayPath": {
+			filter:     bson.D{},
+			projection: bson.D{{"v.0.foo", true}},
```

Comment:
This test ensures that we don't used `doc.SetByPath(...)`. In projection array index path does not find anything.